shard/shard.go

```diff
@@ -49,3 +53,28 @@ type shards struct {
 
 func (s shards) Shards() []Shard { return s.ss }
 func (s shards) ShardsLen() int  { return len(s.ss) }
+
+// ShardingInfo is a set of information for a service's sharding
+type ShardingInfo interface {
```

Comment:
How about remove the Shards info from ServiceInstances, and return the ServiceInstance-Shards data here in the ShardingInfo, this way ServiceInstances can be directly used by non-sharded services? 

And have a WatchShardingInfo function to subscribe to ShardingInfo updates?